core/core.go

```diff
@@ -384,6 +384,21 @@ func (c *Core) CurrentHeader() *types.Header {
 	return c.sl.hc.CurrentHeader()
 }
 
+// CurrentLogEntropy returns the logarithm of the total entropy reduction since genesis for our current head block
+func (c *Core) CurrentLogEntropy() *big.Int {
+	return c.engine.TotalLogS(c.sl.hc.CurrentHeader())
+}
+
+// TotalLogS returns the total entropy reduction if the chain since genesis to the given header
+func (c *Core) TotalLogS(header *types.Header) *big.Int {
+	return c.engine.TotalLogS(header)
+}
+
+// CalcOrder returns the order of the block within the hierarchy of chains
+func (c *Core) CalcOrder(header *types.Header) (*big.Int, int, error) {
```

Comment:
Adding to this, is there a way to not return the intrinsic in a function Calculating the Order of the block

Reply:
> Ideally we should return a common.Order (I think is the correct name) type so we don't have to cast as much later. Also would like a comment to specify what the additional *big.Int return value represents.

@Djadih where do we cast order? To my knowledge, it is only ever used in boolean comparisons and array indexing, both of which require the `int` type.

> is there a way to not return the intrinsic in a function Calculating the Order of the block

@gameofpointers why? We have to calculate the intrinsic to calc the order, and there are several places where we need the intrinsic soon after calculating order. It would be silly to call `CalcOrder` and then subsequently call `CalcIntrinsic` when `CalcOrder` already calculated the intrinsic for you.

What problem does it cause?

Reply:
It doesnt cause any problem, but the function needs a different name

Reply:
Why? `CalcOrder()` calculates the order and returns it as well as the intermediate value used to determine the order. Name seems fine to me :shrug:

A function name should say what the function does. I don't think a function name should fully describe its inputs and outputs.